scoap3/articles/admin.py

```diff
@@ -84,6 +89,38 @@ def article_publisher(self, obj):
     def article_journal(self, obj):
         return [info.journal_title for info in obj.article.publication_info.all()]
 
+    @admin.action(description="Export Selected Articles as CSV")
+    def export_as_csv(self, request, queryset):
+        meta = self.model._meta
+        base_url = f"{request.scheme}://{request.get_host()}"
+        field_names_mapping = {
+            "DOI": "article_doi",
+            "Journal": "article_journal",
+            "Check License": "check_license",
+            "Check File Formats": "check_file_formats",
+            "Check Arxiv Category": "check_arxiv_category",
+            "Check Article Type": "check_article_type",
+        }
+
+        response = HttpResponse(content_type="text/csv")
+        response["Content-Disposition"] = f"attachment; filename={meta}.csv"
+        writer = csv.writer(response)
+        field_names = list(field_names_mapping.keys())
+        rows_titles = ["Link to Article"] + field_names
+        writer.writerow(rows_titles)
+
+        for obj in queryset:
+            article_doi = self.article_doi(obj)
+            article_journal = self.article_journal(obj)
+            values = [
+                getattr(obj, field_names_mapping.get(field, field), None)
+                for field in field_names
+            ]
+            values = [value for value in values if value is not None]
+            row = [f"{base_url}/{article_doi}"] + article_doi + article_journal + values
```

Comment:
better construct the url with provided method (eg `reverse`)

Reply:
can we do it? Since this url is provided by Next.js not by Django, so the url  (as an example: https://backend.dev.scoap3.org/records/26925)  is not in the urlpatterns.

Reply:
ok. true. does it return correctly?

Reply:
base URL returns locally for me the URL that Django server runs, so `http://localhost:8000`, so in the prod it will return `https://backend.dev.scoap3.org/`

Reply:
ok lets write tests